ANSWERS.md

```diff
@@ -0,0 +1,8 @@
+1. PropTypes are used for Typechecking. Typechecking is a way to catch erorrs and rule out errors that you can possibly get. This is important because as
+an application scales more bugs have the chance of being introduced and Typechecking can reduce the amount of possible errors.
+
+2. When developing in React every Component follows a cycle of Mounting, Updating, and Unmounting this is called the Component life-cycle. During this cycle there are many methods available for you to use during the phases. componentDidMount is a useful life-cycle method that is called during the Mouting phase of the application. componentDidMount is only called once in this life-cycle and is typically used for setting state using data from a API call.
+
+3. A Higher Order Component is a advanced technique in React for reusing component logic amongst passed in Components.
+
```

Comment:
Yes, and what is the function of that? Specifically, a Higher Order Component will allow you to extend the functionality of a component or share common functionality between components without repeating code. A Higher Order Component is a misnomer. It is not actually a component, it is a function. A Higher Order Component function takes a component as an argument and returns a component. It transforms a component into another component with more functionality or data. In addition to Authentication the connect function in Redux and the withRouter function from React router are both commonly used HOC function that you will run into in this course.